src/common/random_engine.cpp

```diff
@@ -0,0 +1,36 @@
+#include "common/random_engine.h"
+
+#include <random>
+
+#include "common/assert.h"
+#include "pcg_random.hpp"
+
+namespace kuzu {
+namespace common {
+
+struct RandomState {
```

Comment:
Define the struct inside random_engine.h so you don't have to forward declare in .h file.
And we always put fields in front of the the methods in a struct.